source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -66,7 +67,19 @@ ActiveQuicListener::ActiveQuicListener(Event::Dispatcher& dispatcher,
       crypto_config_.get(), quic_config, &version_manager_, std::move(connection_helper),
       std::move(alarm_factory), quic::kQuicDefaultConnectionIdLength, parent, *config_, stats_,
       per_worker_stats_, dispatcher, listen_socket_);
-  quic_dispatcher_->InitializeWithWriter(new EnvoyQuicPacketWriter(listen_socket_));
+
+  // Create udp_packet_writer
+  udp_packet_writer_ = listener_config.udpPacketWriterFactory()->createUdpPacketWriter(
+      udp_listener_->ioHandle(), listener_config.listenerScope());
+  if (udp_packet_writer_->isBatchMode()) {
+    quic_dispatcher_->InitializeWithWriter(
+        dynamic_cast<Quic::UdpGsoBatchWriter*>(udp_packet_writer_.get()));
```

Comment:
use release() here directly.

nit: please comment why we can do down cast here.

Reply:
Using release() directly here. Updated the comment as well.